From 7fd2ddd4d2233977451d98667f2dd4cb2a5be8d3 Mon Sep 17 00:00:00 2001
From: Jakub Palider <jpalider@marvell.com>
Date: Thu, 11 Feb 2021 03:53:17 +0530
Subject: [PATCH 1253/1921] misc: bphy: Add ioctl to get max irq number

This patch aims at providing users of this driver with
maximum IRQ number via a ioctl call.

Signed-off-by: Jakub Palider <jpalider@marvell.com>
Change-Id: I9165517f5e056217f51cda43261810ac3e718de7
Reviewed-on: https://sj1git1.cavium.com/c/IP/SW/kernel/linux/+/45891
Tested-by: sa_ip-sw-jenkins <sa_ip-sw-jenkins@marvell.com>
Reviewed-by: Radha Chintakuntla <radhac@marvell.com>
Reviewed-by: Sunil Kovvuri Goutham <Sunil.Goutham@cavium.com>
[WK: The original patch got from Marvell sdk11.21.09]
Signed-off-by: Wenlin Kang <wenlin.kang@windriver.com>
---
 drivers/misc/otx_bphy_ctr.c | 8 ++++++++
 1 file changed, 8 insertions(+)

diff --git a/drivers/misc/otx_bphy_ctr.c b/drivers/misc/otx_bphy_ctr.c
index 07591cbb28df..9ea226969327 100644
--- a/drivers/misc/otx_bphy_ctr.c
+++ b/drivers/misc/otx_bphy_ctr.c
@@ -53,6 +53,9 @@ struct otx_irq_usr_data {
 #define OTX_IOC_CLR_BPHY_HANDLER \
 	_IO(OTX_IOC_MAGIC, 2)
 
+#define OTX_IOC_GET_BPHY_MAX_IRQ \
+	_IOR(OTX_IOC_MAGIC, 3, u64)
+
 static inline int __install_el3_inthandler(unsigned long irq_num,
 					   unsigned long sp,
 					   unsigned long cpu,
@@ -155,6 +158,11 @@ static long otx_dev_ioctl(struct file *f, unsigned int cmd, unsigned long arg)
 		if (ret != 0)
 			return -ENOENT;
 		break;
+	case OTX_IOC_GET_BPHY_MAX_IRQ:
+		irq_num = MAX_IRQ;
+		if (copy_to_user((u64 *)arg, &irq_num, sizeof(irq_num)))
+			return -EFAULT;
+		break;
 	default:
 		return -ENOTTY;
 	}
-- 
2.31.1

